### ISSUE TITLE
[feature] bitcoin-cli log

### ISSUE THREAD

**Issue by Sjors** [bitcoin/bitcoin#15438]
The location of log files depends on the operating system and on which network you're using.

It would be nice to have a shortcut which wraps `tail [-f] <datadir>/<network>/debug.log`. It would do what `bitcoind -printtoconsole` does now, but for `bitcoin-cli` so you don't have to restart.

A simpler alternative is a command that returns the log path, e.g.:

```
tail -f `bitcoin-cli logpath`
```

**Comment by darosior** [comment]
Which command outputs should / should not be logged ?
Should we log commands too ?

**Comment by Sjors** [comment]
@darosior this doesn't change what we already log, it's a shortcut to the existing debug.log

**Comment by darosior** [comment]
@Sjors sorry I didn't read carefully enough. I think it's a good idea and I will submit a PR

**Comment by ryanofsky** [comment]
I think it'd be better to add this to an existing get*info RPC method instead of having a new method, so the overall RPC interface and implementation could be simpler, and the information might be more discoverable. Out of getblockchaininfo, getnetworkinfo, getrpcinfo, and getwalletinfo, `getrpcinfo` might make the most sense.

**Comment by Sjors** [comment]
@ryanofsky if we merely return the log path, then I agree. Though maybe it does require a new call like `getprocessinfo`, especially if we end up with potentially multiple log files in #10102?

**Comment by ryanofsky** [comment]
I wouldn't want a `getprocessinfo` to be added for #10102. It is true that in #10102, the `bitcoin-node` RPC interface is still used to access wallets. But this is just for backwards compatibility. After #10102, I think it would make more sense for wallet processes to listen on their own RPC ports, and for the node process not to be involved at all starting wallets, stopping wallets, forwarding RPC requests to wallets, or returning any information about them in it's own RPC interface.

**Comment by Sjors** [comment]
Ok, then adding a `logpath` entry to `getrpcinfo` seems reasonable.

**Comment by promag** [comment]
Why get this via RPC? Could just run, let's say `bitcoind -testnet -printconfig`?

**Comment by darosior** [comment]
Or another command like `bitcoin-cli getconfig`.
But the initial behaviour wanted was about logs : 
> It would be nice to have a shortcut which wraps `tail [-f] <datadir>/<network>/debug.log`. It would do what `bitcoind -printtoconsole` does now, but for `bitcoin-cli` so you don't have to restart.

**Comment by promag** [comment]
> Or another command like `bitcoin-cli getconfig`.

That would be a RPC command. I'm trying to understand if running the above (example) command would be enough.

**Comment by darosior** [comment]
Yes but it is precisely about not using `bitcoind`. Why do you think we should not use RPC ?

**Comment by promag** [comment]
Well I just think it sounds bad - but a RPC `getconfig` would be better than adding to `getrpcinfo` IMO.

Having, for instance, `bitcoind -printconfig` sounds cool because it would work even if you don't have the daemon running. It would also work if you specify other args like `-testnet`. It would be a cool way to evaluate config changes.

**Comment by darosior** [comment]
I think a `getconfig` command could be helpful, but do you have an idea of what should it returns in addition to the `logpath` or (`datadir`) ? But it would be aside from a `getlogs` command (which I still think could be helpful too) and we maybe should discuss it in another issue ?

**Comment by Sjors** [comment]
I like to avoid the need to restart `bitcoind`.

If `bitcoind -testnet -printconfig` works without a restart and can return something easily parseable like JSON, then that's probably fine.

**Comment by promag** [comment]
> I like to avoid the need to restart `bitcoind`.
> 
> If `bitcoind -testnet -printconfig` works without stopping an existing node and can return something easily parseable like JSON, then that's probably fine.

Yes, it works with `bitcoind` running.

**Comment by Sjors** [comment]
Fixed by #15483.
